test/al_log_filter_test.js

```diff
@@ -48,11 +65,83 @@ describe('Unit Tests', function() {
             return done();
         });
         
-        it('Undefined filters', function(done) {
+        it('Undefined filters', function (done) {
             assert.deepEqual(msgString, alLogFilter.filterRegExp(msgString, null));
             assert.deepEqual(msgJson, alLogFilter.filterJson(msgJson));
             return done();
         });
-        
+
+        it('Should filter array based on object child property', function (done) {
+            assert.deepEqual([{
+                messageB: {
+                    childTestMsg: 'childTest',
+                    childTestValue: 'childValue'
+                },
+                text: 'testb'
+            }], alLogFilter.filterJson(msgJson, '{"messageB":{"childTestMsg":"childTest"}}'));
+            return done();
+        });
+
+        it('Should filter array based on object child of child property', function (done) {
+            assert.deepEqual([{
+                messageC: {
+                    childTestMsg: 'childTest',
+                    childTestValue: {
+                        messageC: "c"
+                    }
+                },
+                text: 'testb'
+            }], alLogFilter.filterJson(msgJson, '{"messageC":{"childTestMsg": "childTest","childTestValue":{"messageC":"c"}}}'));
+            return done();
+        });
+
+        it('Should filter array based on array of object with AND case', function (done) {
```

Comment:
I think we agreed that an array of filters always works as `OR`, no?
`AND` is covered with specifying multiple properties in one filter object.

Reply:
Done